### DISCUSSION TITLE
[Keyball39]左手側であることを認識しない、1つだけキーを認識しない

### DISCUSSION THREAD

**Discussion by UuuziI** [Yowkees/keyball#212] [Q&A]
ご助言いただけると幸いです。
2点問題が起きております。
①左手であることを認識しない
トラックボール無し側を左手として使用する予定です。
こちらの基盤のみLEFTのジャンパをさせていますが、右手として認識されます。
何度かはんだ付けをし直しましたが、変化がない状況です。
原因がわからず、途方に暮れております。
![Remap](https://user-images.githubusercontent.com/130230380/230725421-78205a30-3fca-4b87-a54e-46b91596ca1c.png)

②1つだけキーを認識しない
親指ロープロファイルで使用する予定です。
キーソケットのはんだ付け不良かと思い、再はんだ付けや、キースイッチの交換、ダイオードの交換を実施しましたが、変化なしです。
キーソケットの両端子をピンセットでショートさせても、入力できませんでした。（CherryMX互換ソケットも同様）
テスターを持っていないので、キーソケットの導通確認はできておりません。
テスターを購入した場合、どのように導通確認すれば原因の特定が可能でしょうか。
![InkedLeft](https://user-images.githubusercontent.com/130230380/230724586-80a27f1c-205c-4f29-82cb-ea1be540e58d.jpg)

宜しくお願い致します。

**Comment by Yowkees** [comment]
ご連絡が遅くなり申し訳ありませんでした。
２つの問題に共通点がありまして、どちらもCOL5という信号線を使用しています。
下記画像の緑矢印で示した端子になります。
この端子のProMicro側を半田付けし忘れているということはありませんか？
![Keyball39_Left](https://user-images.githubusercontent.com/84209772/231068204-a5bf81ae-c7cb-4473-b035-344c76400f3d.jpg)
お手数ですがご確認お願いします。

**Reply by UuuziI** [reply]
ご連絡いただきありがとうございます。
はんだ付けはしてありましたが、念のため再度はんだ付けをしましたが変化はありませんでした。
が、コンスルーピンがを何度かつけなおしたところ、左側の認識＆キーの認識をするようになりました！
コンスルーピンとProMicroを少し斜めにはんだ付けしていたので、接触が悪かったことが原因のようです。
現在も少し触ってしまうと、もとに戻ってしまう時もあります。
強く押し込むと基本的には正常に認識できています。

遂に完成と思ったのも束の間、さらなる問題が発生しました…
USBを差している側しか、文字入力ができない状態です。
OLED上はUSBがない側はロゴが出ているので適切だと思うのですが、キー入力を一切認識ませんでした。
Firmwareも再度Flashを左右にかけましたが、変化なしです。
何度もお手間を取らせてしまい申し訳ございませんが、次の対応をご教示いただけますと幸いです。
宜しくお願い致します。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
左右の通信に問題があるので、ProMicroのDATA線の半田付けを見直してください。ProMicro上にRXIと表記されている端子です。

また、TRSコネクタの半田付けが甘くても通信不良の可能性があります。半田が基板のパッドまで届いていない場合があり、きちんと山形に半田が盛られているか確認ください。

このあたりも写真があるとアドバイスできるかもしれません。
お手数ですがご対応お願いいたします

**Comment by UuuziI** [comment]
ProMicroのはんだ付けがいまいちだったのがすべての原因だったようで、無事に使えるようになりました!
丁寧なアドバイス、ありがとうございました。
これからKeyballを極めていきます!!
